### ISSUE TITLE
Addition of Advanced Rate Limiting Resources

### ISSUE THREAD

**Issue by syedhusainovo** [cloudflare/terraform-provider-cloudflare#2433]
### Current Terraform and Cloudflare provider version

Terraform v1.4.6
on darwin_amd64

### Description

I would like to request the addition of advanced rate limiting resources to the terraform-provider-cloudflare repository. Currently, the available resources for rate limiting in Terraform are limited to basic configurations. Introducing advanced rate limiting resources would greatly enhance the capabilities of the Cloudflare Terraform provider and provide significant benefits to users.

Including advanced rate limiting resources in the terraform-provider-cloudflare would align the provider with the full range of rate limiting options available through Cloudflare's API. This feature would empower users to manage rate limiting configurations in a unified manner, streamlining their workflows and reducing the need for manual intervention or configuration changes outside of Terraform.

Thank you for considering this feature request. I believe that the addition of advanced rate limiting resources would greatly enhance the value of the terraform-provider-cloudflare and provide users with powerful capabilities to protect and manage their applications effectively.

### Use cases

The end goal is to have more fine-grained control over rate limiting configurations in Terraform for Cloudflare. Currently, the available rate limiting resources in terraform-provider-cloudflare offer basic functionality, but they do not support advanced configurations. As a result, users face limitations when trying to implement complex rate limiting strategies tailored to their specific application's needs.

The problem is that the existing rate limiting resources lack the flexibility to handle scenarios where rate limiting rules need to be based on specific criteria such as IP addresses, HTTP headers, user agents, or custom attributes. These criteria are crucial for accurately identifying and categorizing incoming traffic and applying appropriate rate limiting policies.

By introducing advanced rate limiting resources, users will be able to define more granular rate limiting rules that align with their unique use cases. This includes the ability to set rules based on specific attributes or conditions and to create sophisticated rate limiting policies that balance the need for security, performance, and accessibility.

In summary, the end goal is to enable users to have greater control and customization over rate limiting configurations in Terraform for Cloudflare. This will address the limitations of the current basic rate limiting resources and empower users to implement advanced rate limiting strategies that effectively protect their applications while allowing legitimate traffic to flow unhindered.

### Potential Terraform configuration

```hcl
...
```

### References

https://developers.cloudflare.com/waf/rate-limiting-rules/

**Comment by jacobbednarz** [comment]
there is already support for the rules engine based rate limiting via `cloudflare_ruleset`. the `cloudflare_rate_limit` is the legacy resource.
